File: validator-app/src/main/java/org/cna/keyple/famoco/validator/di/UIModule.java
package org.cna.keyple.famoco.validator.di;

import org.cna.keyple.famoco.validator.di.scopes.ActivityScoped;
import org.cna.keyple.famoco.validator.di.scopes.FragmentScoped;
import org.cna.keyple.famoco.validator.ui.activities.MainActivity;
import org.cna.keyple.famoco.validator.ui.fragments.CardReaderFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class UIModule {
    @ActivityScoped
    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

    @FragmentScoped
    @ContributesAndroidInjector
    abstract CardReaderFragment cardReaderFragment();
}

File: validator-app/src/main/kotlin/org/cna/keyple/famoco/validator/di/UIModule.kt
package org.cna.keyple.famoco.validator.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import org.cna.keyple.famoco.validator.di.scopes.ActivityScoped
import org.cna.keyple.famoco.validator.di.scopes.FragmentScoped
import org.cna.keyple.famoco.validator.ui.activities.MainActivity
import org.cna.keyple.famoco.validator.ui.fragments.CardReaderFragment

@Module
abstract class UIModule {
    @ActivityScoped
    @ContributesAndroidInjector
    abstract fun mainActivity(): MainActivity

    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun cardReaderFragment(): CardReaderFragment
}